packages/overlay/src/integrations/sentry/components/insights/agents/AITraceDetails.tsx

```diff
@@ -0,0 +1,345 @@
+
+  if (!span) {
+    return (
+      <div className="p-6">
+        <p className="text-red-400">Trace not found</p>
+        <p className="text-sm text-gray-400">Looking for span ID: {spanId}</p>
+      </div>
+    );
+  }
+
+  const trace = createAITraceFromSpan(span);
+
+  if (!trace) {
+    return (
+      <div className="p-6">
+        <p className="text-red-400">Unable to process AI trace</p>
+        <p className="text-sm text-gray-400">Span ID: {spanId}</p>
+        <p className="text-sm text-gray-400">Span description: {span.description}</p>
+      </div>
+    );
+  }
+
+  const handler = detectAILibraryHandler(span);
+
+  if (!handler) {
+    return (
+      <div className="p-6">
+        <p className="text-red-400">No Spotlight AI handler found for this AI trace</p>
+        <p className="text-sm text-gray-400">Span ID: {spanId}</p>
+        <p className="text-sm text-gray-400">Span description: {span.description}</p>
+      </div>
+    );
+  }
+
+  const traceContext = { trace_id: span.trace_id };
+  const startTimestamp = span.start_timestamp;
+  const totalDuration = span.timestamp - span.start_timestamp;
+
+  return (
+    <SearchProvider>
+      <div className="h-full overflow-y-auto">
+        {/* Header Section */}
+        <div className="border-b-primary-700 bg-primary-950 border-b px-6 py-4">
+          <div className="mb-2 flex items-center gap-2">
+            <h2 className="text-xl font-bold">{handler.getDisplayTitle(trace)}</h2>
+          </div>
+          <div className="flex items-center gap-2">
+            <span className="text-primary-400 text-sm">{trace.id}</span>
+            <Badge color="primary">{handler.getTypeBadge(trace)}</Badge>
+            {trace.metadata.modelId && <Badge color="secondary">{trace.metadata.modelId}</Badge>}
+            {trace.metadata.functionId && <Badge color="neutral">{trace.metadata.functionId}</Badge>}
+          </div>
+        </div>
+
+        {/* Content */}
+        <div className="space-y-6 p-6">
+          <div>
+            <div className="flex flex-col space-y-4">
+              <div className="text-primary-300 flex flex-1 items-center gap-x-1">
+                <DateTime date={span.start_timestamp} />
+                <span>&mdash;</span>
+                <span>
+                  <strong>{getFormattedDuration(span.timestamp - span.start_timestamp)}</strong> duration
+                </span>
+              </div>
+              <div className="flex-1">
+                <div className="border-primary-800 relative h-8 border py-1">
+                  <div
+                    className="bg-primary-800 absolute bottom-0 top-0 -m-0.5 flex w-full items-center p-0.5"
+                    style={{
+                      left: 0,
+                      width: "100%",
+                    }}
+                  >
+                    <span className="whitespace-nowrap">
+                      {getFormattedDuration(span.timestamp - span.start_timestamp)}
```

Comment:
You already have `totalDuration` defined above?
```suggestion
                      {getFormattedDuration(totalDuration)}
```